### PR TITLE
Favorite button & Return button

### PR DESCRIPTION
**Pull request by brigidoeduardo**
Details Activity Return Button Set (xml + finish)
Details Activity Favorite Button Toggle Set (xml + drawables)

### CODE REVIEW

**Review by Canato — CHANGES_REQUESTED** [review]
Good PR

File: app/build.gradle [hunk 1]
```diff
@@ -42,6 +42,7 @@ dependencies {
     implementation 'com.squareup.retrofit2:converter-moshi:2.8.1'
     implementation 'com.squareup.okhttp3:okhttp:4.11.0'
     implementation 'com.squareup.okhttp3:logging-interceptor:4.11.0'
+    implementation 'androidx.room:room-ktx:2.6.1'
```
[the remark this answers is not in the document]
Comment (brigidoeduardo) [reply app/build.gradle:45]: Done

File: app/src/main/java/com/monzo/androidtest/articledetails/data/ArticleDetailsRepository.kt [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
 ) {
-    fun getArticle(articleUrl: String,): Single<ApiArticle> =
+    private val sharedPreferences: SharedPreferences by lazy {
+        context.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
+    }
+
+    companion object {
+        private const val SHARED_PREF_NAME = "ArticleDetailsPrefs"
+        private const val KEY_FAVORITE_STATE_PREFIX = "favorite_state_"
+        private const val KEY_FAVORITE_ARTICLES = "favorite_articles"
```
[the remark this answers is not in the document]
Comment (brigidoeduardo) [reply app/src/main/java/com/monzo/androidtest/articledetails/data/ArticleDetailsRepository.kt:20]: Done

File: app/src/main/java/com/monzo/androidtest/articledetails/presentation/ArticleDetailsViewModel.kt [hunk 3]
```diff
@@ -7,10 +7,10 @@ import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
 
 class ArticleDetailsViewModel(
-    private val repository: ArticleDetailsRepository,
-    private val articleUrl: String
+    val repository: ArticleDetailsRepository,
+    val articleUrl: String
```
[the remark this answers is not in the document]
Comment (brigidoeduardo) [reply app/src/main/java/com/monzo/androidtest/articledetails/presentation/ArticleDetailsViewModel.kt:11]: The private was greyed out, so I removed it, in the new IDE is back to being highlighted, all good

File: app/src/main/java/com/monzo/androidtest/articledetails/data/ArticleDetailsRepository.kt [hunk 4]
```diff
@@ -1,14 +1,45 @@
...
+    }
+
+    fun getArticle(articleUrl: String): Single<ApiArticle> =
         guardianService.getArticle(articleUrl, "main,body,headline,thumbnail")
             .map { it.response.content }
-}
+
+
+    fun getAllFavoriteArticles(): Set<String> {
+        return sharedPreferences.all.keys.filter { it.startsWith(KEY_FAVORITE_STATE_PREFIX) }
```
[the remark this answers is not in the document]
Comment (brigidoeduardo) [reply app/src/main/java/com/monzo/androidtest/articledetails/data/ArticleDetailsRepository.kt]: Done

File: app/src/main/java/com/monzo/androidtest/articledetails/data/ArticleDetailsRepository.kt [hunk 5]
```diff
@@ -1,14 +1,45 @@
...
 import io.reactivex.Single
 
 class ArticleDetailsRepository(
     private val guardianService: GuardianService,
+    private val context: Context
 ) {
-    fun getArticle(articleUrl: String,): Single<ApiArticle> =
+    private val sharedPreferences: SharedPreferences by lazy {
+        context.getSharedPreferences(SHARED_PREF_NAME, Context.MODE_PRIVATE)
+    }
```
[the remark this answers is not in the document]
Comment (brigidoeduardo) [reply app/src/main/java/com/monzo/androidtest/articledetails/data/ArticleDetailsRepository.kt]: Done

**Review by Canato — APPROVED** [review]

**Pull request merged**